src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -520,12 +520,21 @@ protected override Match<SyntaxNode> ComputeBodyMatch(SyntaxNode oldBody, Syntax
 
                 // We use the parent node as a root:
                 // - for field/property initializers the root is EqualsValueClause. 
-                // - for expression-bodies the root is ArrowExpressionClauseSyntax. 
+                // - for member expression-bodies the root is ArrowExpressionClauseSyntax.
                 // - for block bodies the root is a method/operator/accessor declaration (only happens when matching expression body with a block body)
                 // - for lambdas the root is a LambdaExpression.
                 // - for query lambdas the root is the query clause containing the lambda (e.g. where).
+                // - for local functions the root is LocalFunctionStatement.
 
-                return new StatementSyntaxComparer(oldBody, newBody).ComputeMatch(oldBody.Parent, newBody.Parent, knownMatches);
+                SyntaxNode GetMatchingRoot(SyntaxNode body)
+                {
+                    var root = body.Parent;
+                    // We could apply this change across all ArrowExpressionClause consistently not just for ones with LocalFunctionStatement parents
+                    // but it would require an essential refactoring, and not reasonable for now. 
```

Comment:
>, and not reasonable for now [](start = 68, length = 28)

I'd omit ", and not reasonable for now" -- the comment will live in the code for some time...

Reply:
Thank you!

---
In reply to: [145001709](https://github.com/dotnet/roslyn/pull/22624#discussion_r145001709) [](ancestors = 145001709)